distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -151,5 +183,122 @@ private Response respond(Response response) {
+             * advertising supporting byte offsets specifically, so this function assumes that byte
+             * ranges are provided and does not evaluate the unit of measurement
+             *
+             * @param rangeHeaderValue          Value of the range header sent in the request
+             * @param totalSizeOfProductInBytes Total size of the message to be returned. This is
+             *                                  not the partial content size, but the FULL size of
+             *                                  the product.
+             */
+            public ByteRange(String rangeHeaderValue, int totalSizeOfProductInBytes) {
+                // extract bytes
+                String startToken = StringUtils.substringBetween(rangeHeaderValue, "=", "-");
+                String endToken = StringUtils.substringAfter(rangeHeaderValue, "-");
+
+                // range offsets can be blank to indicate "until beginning" or "until end" of data.
+                if (StringUtils.isBlank(startToken)) {
+                    start = 0;
+                } else {
+                    start = Integer.parseInt(startToken);
+                }
+                if (StringUtils.isBlank(endToken)) {
+                    end = totalSizeOfProductInBytes - 1;
+                } else {
+                    end = Integer.parseInt(endToken);
+                }
+            }
+
+            public String contentRangeValue() {
+                return "bytes " + start + "-" + end + "/" + responseMessage.length;
+            }
+        }
+    }
+
+    public static class ChunkedContentBuilder {
+        private String message;
+
+        private Duration delay = Duration.ofMillis(0);
+
+        private int fails = 0;
+
+        private HeaderCapture headerCapture = null;
+
+        /**
+         * Set message.
+         *
+         * @param message Message to be sent in the response.
+         */
+        public ChunkedContentBuilder(String message) {
+            this.message = message;
+        }
+
+        /**
+         * Set delay between sending each character of the message in milliseconds.
+         *
+         * @param delay Time to wait between sending each character of the message.
+         * @return Builder object
+         */
+        public ChunkedContentBuilder delay(Duration delay) {
+            this.delay = delay;
+            return this;
+        }
+
+        /**
+         * Number of times to fail (simulate network disconnect) after sending
+         * the first character. Once this number is reached, the message will
+         * send successfully.
+         *
+         * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+         *                         the first character. Once this number is reached, the message will
+         *                         send successfully.
+         * @return Builder object
+         */
+        public ChunkedContentBuilder fails(int numberOfFailures) {
```

Comment:
Since the client is telling the builder what to build, `fail` (no _s_) might be a little better here.